my-sms/app/views/institutes/show.html.haml

```diff
@@ -1,3 +1,7 @@
--  display_attrs = %i[id name]
+:ruby
+  display_attrs = %i[id name courses]
+  display_ovrd = {
+  courses: { value: render(partial: 'course', collection: @institute.courses) }
+  }
 = render partial: 'shared/model_attr_list',
-  locals: {model_instance: @institute, display_attrs: display_attrs, display_ovrd: nil}
+  locals: {model_instance: @institute, display_attrs: display_attrs, display_ovrd: display_ovrd}
```

Comment:
Since the last time I saw this, you're now passing another local variable and increasing complexity significantly. You now have to maintain two different variables which highly depend on each other `display_attrs` and  `display_ovrd`.

What is exactly the meaning of this?
```
courses: { value: render(partial: 'course', collection: @institute.courses) }
``
Are you really passing a full HTML inside a Hash? Why did you choose to override the values instead of assigning them from the beginning?

Reply:
This is actually a different partial to the one you are remembering I believe.

Overriding the values allows me to create a single source of truth for the layout structure, also making it easy to reproduce the same view logic for different models and weakly enforces some sort of visual consistency. `display_ovrd` achieves this at the minimal cost of flexibility by allowing the default title and content display to be overridden.

Passing partials into `display_ovrd` is by design as it allows complex view components to be rendered, however, in this case I could achieve the same result without a partial since the course list is pretty basic. The reason I'm using a partial here is to clean up the code here and provide some DRYness in case a list of courses needs to be rendered elsewhere. This is in line with the way Rails is intended to be used IMO, the guide describes using partials to render collections via the `collection` arg
https://guides.rubyonrails.org/v3.2.21/layouts_and_rendering.html#using-partials

There was already some discussion around this in a previous PR and it did have some benefit in later tasks so I would prefer to leave it in.

Reply:
hmm, hang on, you're actually rendering the entire `list` view for coureses for each institute, yes? I think this will result in a lot of db queries (a la the n+1 problem), are you able to confirm the sql calls made when rendering this view?

Reply:
Its just a partial displaying the course name as a link on the individual `institute` show page.
my-sms/app/views/institutes/_course.html.haml

There's no recursion here, rails handles the collection rendering efficientily when using the `collection` arg I believe. I can log the sql calls if you still want me to check.

Reply:
oh cool 🙂 didn't see that partial there. Seems OK to me then as it continues from the pattern used previously